armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -225,8 +225,51 @@ class can then use to modify the isotopics as necessary.
             constructedObject = components.factory(shape, [], kwargs)
             _setComponentFlags(constructedObject, self.flags, blueprint)
             insertDepletableNuclideKeys(constructedObject, blueprint)
+
+        # set the custom density for non-custom material components after construction
+        self.setCustomDensity(constructedObject, blueprint, matMods)
+
         return constructedObject
 
+    def setCustomDensity(self, constructedComponent, blueprint, matMods):
+        """Apply a custom density to a material with custom isotopics but not a 'custom material'."""
+        if self.isotopics is not None:
+            # must nest because one can't check for attribues on a NoneType
+            if blueprint.customIsotopics[self.isotopics].density > 0:
```

Comment:
If I go into the Python interpreter and do this:

```bash
>>> None > 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

If the `.density` is `None`, this line doesn't return `False`, it raises a `TypeError`.

Reply:
I've altered the logic in this section, let me know what you think.